scripts/o3de/o3de/engine_template.py

```diff
@@ -226,50 +259,96 @@ def _execute_restricted_template_json(json_data: dict,
                 restricted_json_data.update({"restricted_name": destination_name})
                 s.write(json.dumps(restricted_json_data, indent=4) + '\n')
 
+    ###################################################################################
+    # for each createDirectories in the template copy any entries in the json_data that are for this platform
+    for create_directory in template_json_data['createDirectories']:
+        new_dir = pathlib.Path(create_directory['dir'])
+        if not keep_restricted_in_instance and 'Platform' in new_dir.parts:
+            try:
+                # the name of the Platform should follow the '/Platform/'
+                pattern = r'/Platform/(?P<Platform>[^/:*?\"<>|\r\n]+/?)'
+                found_platform = re.search(pattern, new_dir.as_posix()).group('Platform')
```

Comment:
Actually this logic should be made into a function that extracts the platform directory, instead of copying and pasting it

```suggestion
def extract_platform_name(input_path : pathlib.Path) -> str or None:
     platform_name = None
     platform_index = input_path.parts.index('Platform')
     if platform_index != -1 and platform_index + 1 < len(input_path.parts):
         platform_name = input_path.parts[platform_index + 1]
     return platform_name
```